### PR TITLE
add toc to spec

### PR DESCRIPTION
**Pull request by kevholmes**
Resolve issue https://github.com/in-toto/docs/issues/3

As I was working my way through the spec I also cleaned up a number of basic markdown linting errors that popped up along the way.

### CODE REVIEW

**Comment by adityasaky** [comment]
This is neat! Let me see if this also applies to the PDF. Also, did you use an automated plugin or something to generate it? Just so we can keep it updated as we go along.

**Comment by adityasaky** [comment]
Hmm, looks like pandoc struggles with deeply nested lists, or rather, it converts to an intermediate latex form that struggles?

File: in-toto-spec.md [hunk 1]
```diff
@@ -2,10 +2,81 @@
 
 Apr 11, 2021
```
Comment (adityasaky) [inline in-toto-spec.md:3]:
```suggestion
Oct 1, 2021
```

**Comment by SantiagoTorres** [comment]
Hmm, I think we can make this happen for now and figure out the pandoc bits afterwards...

**Comment by adityasaky** [comment]
I think I've got it working with an alternative pdf engine. Let me push to this PR and we should be good to go.

**Comment by adityasaky** [comment]
Thanks again, @kevholmes!

**Pull request merged**